plugins/database/redshift/redshift.go

```diff
@@ -229,132 +172,180 @@ func (r *RedShift) CreateUser(ctx context.Context, statements dbplugin.Statement
+				query = strings.TrimSpace(query)
+				if len(query) == 0 {
+					continue
+				}
+
+				m := map[string]string{
+					"name":     username,
+					"username": username,
+					"password": password,
+				}
+				if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
+					return dbplugin.UpdateUserResponse{}, err
+				}
 			}
 		}
 	}
 
-	return tx.Commit()
+	return dbplugin.UpdateUserResponse{}, tx.Commit()
 }
 
-func (r *RedShift) RevokeUser(ctx context.Context, statements dbplugin.Statements, username string) error {
+// DeleteUser supports both default and custom statements to delete a user.
+func (r *RedShift) DeleteUser(ctx context.Context, req dbplugin.DeleteUserRequest) (dbplugin.DeleteUserResponse, error) {
 	// Grab the lock
 	r.Lock()
 	defer r.Unlock()
 
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
-	if len(statements.Revocation) == 0 {
-		return r.defaultRevokeUser(ctx, username)
+	if len(req.Statements.Commands) == 0 {
+		return r.defaultDeleteUser(ctx, req)
 	}
 
-	return r.customRevokeUser(ctx, username, statements.Revocation)
+	return r.customDeleteUser(ctx, req)
 }
 
-func (r *RedShift) customRevokeUser(ctx context.Context, username string, revocationStmts []string) error {
+func (r *RedShift) customDeleteUser(ctx context.Context, req dbplugin.DeleteUserRequest) (dbplugin.DeleteUserResponse, error) {
 	db, err := r.getConnection(ctx)
 	if err != nil {
-		return err
+		return dbplugin.DeleteUserResponse{}, err
 	}
 	defer db.Close()
 
 	tx, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return dbplugin.DeleteUserResponse{}, err
 	}
 	defer func() {
 		tx.Rollback()
 	}()
 
-	for _, stmt := range revocationStmts {
+	for _, stmt := range req.Statements.Commands {
 		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
 			query = strings.TrimSpace(query)
 			if len(query) == 0 {
 				continue
 			}
 
 			m := map[string]string{
-				"name": username,
+				"name": req.Username,
 			}
 			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
-				return err
+				return dbplugin.DeleteUserResponse{}, err
 			}
 		}
 	}
 
-	return tx.Commit()
+	return dbplugin.DeleteUserResponse{}, tx.Commit()
 }
 
-func (r *RedShift) defaultRevokeUser(ctx context.Context, username string) error {
+func (r *RedShift) defaultDeleteUser(ctx context.Context, req dbplugin.DeleteUserRequest) (dbplugin.DeleteUserResponse, error) {
 	db, err := r.getConnection(ctx)
 	if err != nil {
-		return err
+		return dbplugin.DeleteUserResponse{}, err
 	}
 	defer db.Close()
 
+	username := req.Username
+
 	// Check if the role exists
 	var exists bool
 	err = db.QueryRowContext(ctx, "SELECT exists (SELECT usename FROM pg_user WHERE usename=$1);", username).Scan(&exists)
 	if err != nil && err != sql.ErrNoRows {
-		return err
+		return dbplugin.DeleteUserResponse{}, err
 	}
 
 	if !exists {
-		return nil
+		return dbplugin.DeleteUserResponse{}, nil
```

Comment:
Can you put a comment on this saying that we don't return an error because redshift may have deleted the user via TTL before we got to it?